src/connection/incoming.js

```diff
@@ -68,9 +68,6 @@ class IncomingConnectionFSM extends BaseConnection {
       this.emit('muxed', this.conn)
     })
     this._state.on('DISCONNECTING', () => {
-      if (this.theirPeerInfo) {
-        this.theirPeerInfo.disconnect()
-      }
```

Comment:
This was unnecessary. Incoming connections don't cause the connect to happen until after the are muxed, which will now be handled by the `switch.connection.add()` logic.